src/SeleniumLibrary/keywords/window.py

```diff
@@ -161,19 +161,29 @@ def maximize_browser_window(self):
         self.driver.maximize_window()
 
     @keyword
-    def get_window_size(self):
+    def get_window_size(self, inner=False):
         """Returns current window width and height as integers.
 
         See also `Set Window Size`.
 
+        With optional `inner` parameter you receive
+        the inner size, excluding the browser bars, borders and so on.
+
         Example:
         | ${width} | ${height}= | `Get Window Size` |
+        | ${width} | ${height}= | `Get Window Size` | True |
         """
-        size = self.driver.get_window_size()
-        return size['width'], size['height']
+        if inner == True:
+            inner_width = int(self.driver.execute_script("return window.innerWidth;"))
+            inner_height = int(self.driver.execute_script("return window.innerHeight;"))
+
+            return inner_width, inner_height
+        else:
+            size = self.driver.get_window_size()
+            return size['width'], size['height']
 
     @keyword
-    def set_window_size(self, width, height):
+    def set_window_size(self, width, height, inner=False):
```

Comment:
We need to design how the keyword should work with frames. If frame is selected, `window.innerWidth` and `window.innerHeight` returns the frame size. Logically thinking, it would be best to return to a main/outermost windows/frame (In practice `self.driver.switch_to.default_content()` would be needed.) If do that, then it is not possible to return to a frame which is was selected, because Selenium or SeleniumLibrary does not track nested frame is selected. But because this is new functionality, when `inner=True` then it is OK to do: `self.driver.switch_to.default_content()`

But what we should do with the `get_window_size` method. I feel that, if change this keyword to do: `self.driver.switch_to.default_content()` then same should be done with `get_window_size` method. 

Also these changes must be documented and tested.

Reply:
I understand that. I'm a bit unsure about doing the switch, it could be annoying for the user. Leaving the frame sizing could be "reasonable" because of the literally word meaning of the parameter "Inner". But I will think about that a bit more, and I will do some tests with frames.

Reply:
After thinking this bit more, I am starting to think that doing `self.driver.switch_to.default_content()` may not be good idea. It will strip's the possibility from the user to control what frame will be resized. If we just trust the user to select correct frame and resized the selected frame it will bring more control for the users.

In any way we choose, there must be a testswith frames.

Reply:
Are you sure you want to hijack this Keyword and not create another one? Set Window ViewPort or something like that?  I agree that automatically switching the user to the default content would be "rude" but maybe instead force to switch to default content themselves if that is what they mean by wanting to set the window size vs. wanting to set the "web app view size"?  Just some thoughts.

Reply:
The keyword default functionality is not changed. Only if `inner` argument is true, the functionality is different.